presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/MetadataResolver.java

```diff
@@ -87,4 +90,24 @@ default boolean isMaterializedView(QualifiedObjectName viewName)
     {
         return getMaterializedView(viewName).isPresent();
     }
+
+    /**
+     * Returns list of registered types.
+     */
+    List<Type> getTypes();
+
+    /**
+     * Returns list of registered parametric types.
+     */
+    List<ParametricType> getParametricTypes();
```

Comment:
Do we actually need to expose ParametricType?

Reply:
Why not? We need to support for all types, am I missing something here? let me discuss offline.